llvm/include/llvm/ADT/PagedVector.h

```diff
@@ -0,0 +1,132 @@
+//===- llvm/ADT/PagedVector.h - 'Lazyly allocated' vectors --------*- C++
+//-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the PagedVector class.
+//
+//===----------------------------------------------------------------------===//
+#ifndef LLVM_ADT_PAGEDVECTOR_H
+#define LLVM_ADT_PAGEDVECTOR_H
+
+#include <cassert>
+#include <vector>
+
+namespace llvm {
+// A vector that allocates memory in pages.
+// Order is kept, but memory is allocated only when one element of the page is
+// accessed. This introduces a level of indirection, but it is useful when you
+// have a sparsely initialised vector where the full size is allocated upfront
+// with the default constructor and elements are initialised later, on first
+// access.
+//
+// Notice that this does not have iterators, because if you
+// have iterators it probably means you are going to touch
+// all the memory in any case, so better use a std::vector in
+// the first place.
+template <typename T, std::size_t PAGE_SIZE = 1024 / sizeof(T)>
+class PagedVector {
+  static_assert(PAGE_SIZE > 0, "PAGE_SIZE must be greater than 0. Most likely you want it to be greater than 16.");
```

Comment:
Please run this through clang-format

Reply:
I actually did:

```
# ktf at stillwater-97.local in ~/src/llvm-upstream on git:main ● [17:18:18]
→ git-clang-format --diff 571e4f233ba99c09d5a34d4676c6479ec84dc48a HEAD | patch -p1

# ktf at stillwater-97.local in ~/src/llvm-upstream on git:main ● [17:18:27]
→ git diff

# ktf at stillwater-97.local in ~/src/llvm-upstream on git:main ● [17:18:30]
→ clang-format --version
clang-format version 14.0.4
```

Am I missing something trivial?

Reply:
It is easiest to have everything in a single commit and then do `git-clang-format HEAD~`. Then `git add`.